packages/arb-token-bridge-ui/src/hooks/TransferPanel/useGasEstimates.ts

```diff
@@ -40,44 +53,53 @@ export function useGasEstimates({
   destinationChainId,
   sourceChainErc20Address,
   destinationChainErc20Address,
-  amount
+  amount,
+  balance
 }: {
   walletAddress?: string
   sourceChainId: number
   destinationChainId: number
   sourceChainErc20Address?: string
   destinationChainErc20Address?: string
   amount: BigNumber
+  balance: BigNumber | null
 }): {
   gasEstimates: GasEstimates | DepositGasEstimates | undefined
   error: any
 } {
   const { data: signer } = useSigner()
 
   const { data: gasEstimates, error } = useSWR(
-    typeof signer === 'undefined'
-      ? null
-      : [
-          walletAddress,
+    signer && balance && balance.gt(amount)
+      ? ([
+          signer,
```

Comment:
`signer` is an object and I think it won't be serializable here just like `amount` and `balance`.

Reply:
Changed, good catch